### PR TITLE
Update dev docs to include latest information

### PR DESCRIPTION
**Pull request by mschwager**
This includes: 1) the inclusion of `spacegrep`, and 2) the use of `SEMGREP_SKIP_BIN` to BYOBinaries.

For #2929.

### CODE REVIEW

File: semgrep/README.md [hunk 1]
```diff
@@ -1,17 +0,0 @@
-# semgrep-cli Development
-
-Semgrep uses pipenv to manage a virtual environment:
-
-```
-pipenv install --dev
```
Comment (brendongo) [inline semgrep/README.md:6]: Is this information stored elsewhere now?

File: doc/README.md [hunk 2]
```diff
@@ -103,23 +106,23 @@ Note that in the virtual environment `semgrep` was installed in "editable"
...
+and inside should be the necessary binaries. You can confirm this by running:
 
 ```bash
-$ ./semgrep-core -version
+$ ./semgrep-core --help
+$ ./spacegrep --help
 ```
 
-Copy this file to somewhere in your `$PATH` so `semgrep` can find it. For
+Copy these files to somewhere in your `$PATH` so `semgrep` can find it. For
```
Comment (aryx) [inline doc/README.md]: find them

**Review by aryx — APPROVED** [review]

File: semgrep/README.md (hunk 1 above)
Comment (mschwager) [reply semgrep/README.md:6]: Yeah, it's been here for a long time: https://github.com/returntocorp/semgrep/tree/develop/doc#semgrep-development

Or, at least, that used to be in `CONTRIBUTING.md`, now we link that to `doc/README.md`.

Comment (mschwager) [reply semgrep/README.md:6]: And testing: https://github.com/returntocorp/semgrep/tree/develop/doc#testing

File: doc/README.md (hunk 2 above)
Comment (mschwager) [inline doc/README.md]:
```suggestion
Copy these files to somewhere in your `$PATH` so `semgrep` can find them. For
```

**Pull request merged**